tests/queries/0_stateless/00652_mutations_alter_update.sh

```diff
@@ -123,5 +123,22 @@ ${CLICKHOUSE_CLIENT} --query="SELECT value2, materialized_value FROM alter_updat
 
 ${CLICKHOUSE_CLIENT} --query="ALTER TABLE alter_update DROP PARTITION 200001"
 
+${CLICKHOUSE_CLIENT} --query="SELECT '*** Test updating with SETTINGS update_allow_materialized_columns=1 ***'"
```

Comment:
It fails because InterpreterAlterQuery specifically looks for SETTINGS in the query.
I guess also picking it up from the context would be more consistent, I'll do that.